src/DynamoWebServer/Messages/MessageHandler.cs

```diff
@@ -465,6 +472,54 @@ private string GetValue(NodeModel node)
             return data;
         }
 
+        private object GenerateWatchData(MirrorData data)
+        {
+            if (data.IsCollection)
+            {
+                var list = new List<object>();
+                var elements = data.GetElements();
+
+                foreach (var mirrorData in elements)
+                {
+                    if (mirrorData.IsCollection)
+                    {
+                        list.Add(GenerateWatchData(mirrorData));
+                    }
+                    else if (mirrorData.Data != null)
+                    {
+                        double number;
+                        list.Add(double.TryParse(mirrorData.StringData, out number)
+                            ? mirrorData.Data
+                            : mirrorData.Data.ToString());
+                    }
+                    else
+                    {
+                        list.Add(mirrorData.Class == null
+                            ? "null"
+                            : mirrorData.Class.ClassName);
```

Comment:
@pboyer, additional logic is required to provide same elements view in Flood as in Dynamo. If mirrorData.Data equals to null we need to send class name, and also we need to send number not as a string to have ability to display it correctly in Flood.